### PR TITLE
GAWB-1567: unit and integration tests for ElasticSearch

### PR DESCRIPTION
**Pull request by davidangb**
At a high level, changes in this PR are:
* create example unit tests that validate the requests we send to ES
* create example integration tests that validate the results we get back from ES
* allow integration tests to be run during Jenkins build

as before, unit tests are `sbt test`.

to run integration tests locally, run `sbt it:test`.

Integration tests now run every time we build the docker image (and should fail the docker build if the tests fail). This is how we ensure the integration tests run every time in Jenkins. To test this, do a:
```
docker build --build-arg ESURLS=foo:port,bar:port .
```
but replace "foo:port,bar:port" with the value from `elasticsearch.urls` in firecloud-orchestration.conf.

I would have liked to get far more test coverage in this PR, but the goal of this PR was to create a framework in which tests can be written, not to greatly increase coverage.

Drawback to this PR: `docker build .` no longer works, as it will fail integration tests. A docker build requires the `--build-arg ESURLS=foo:port,bar:port`.

I created GAWB-1680 to track follow-on work.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -36,6 +38,8 @@ RUN sbt compile
 COPY src/main src/main
 COPY src/test src/test
 RUN sbt compile
+RUN sbt test
```
Comment (rushtong) [inline Dockerfile:41]: Do we need to run regular tests on build? I'm not opposed to this in principle, but our normal PR process should catch any `sbt test` failures.

Comment (davidangb) [reply Dockerfile:41]: yeah, we certainly don't have to ... but they're also pretty quick, and will catch any environmental errors (works locally, fails in docker, or whatever) so I thought I'd toss 'em in.

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/ElasticSearchDAOQuerySupportSpec.scala [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    }
+  }
+
+  // TODO: does text search criteria properly search against _all?
+  // TODO: does empty text search do the right thing?
+  // TODO: do facet selections properly become term filters?
+  // TODO: do facet requests properly become aggregations?
+  // TODO: does an expanded facet properly expand?
+  // TODO: does pagination work?
+  // TODO: does page size work?
```
Comment (rushtong) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/ElasticSearchDAOQuerySupportSpec.scala:60]: New stories for these TODOs?

**Review by rushtong — APPROVED** [review]
👍 LG, WG. Minor comment inline. Very nice to see this framework in place. I have some ES unit tests in consent that should likely move to integration tests following a similar architecture.

File: Dockerfile (hunk 1 above)
Comment (rushtong) [reply Dockerfile:41]: 👍

**Pull request merged**